parlai/mturk/core/dev/mturk_manager.py

```diff
@@ -1195,23 +1191,32 @@ def _wait_for_task_expirations(self):
         """
         start_time = time.time()
         min_wait = self.opt['assignment_duration_in_seconds']
+        wait_time = 0.1
         while time.time() - start_time < min_wait and len(self.hit_id_list) > 0:
-            self.expire_all_unassigned_hits()
-            time.sleep(max(self.opt['assignment_duration_in_seconds'] / 60, 0.1))
+            for hit_id in self.hit_id_list.copy():
+                self.check_hit_status(hit_id)
+            wait_time *= 1.5
```

Comment:
interesting. why do you wait longer and longer each time?

Reply:
I'm using [Exponential Backoff](https://en.wikipedia.org/wiki/Exponential_backoff) to prevent overloading the MTurk API's request system. Being stuck here means the assignment is waiting in someone's task queue, so there's no good way of knowing when it'll expire. The previous version would sometimes exceed the rate limit.